lessons/advanced/gen-stage.md

```diff
@@ -16,46 +16,353 @@ So what is GenStage?  From the official documentation, it is a "specification an
 
 What it means is that GenStage provides a way for us to define a pipeline of work to be carried out by independent steps (or stages) in a separate processes; if you've worked with pipelines before then some of these concepts should be familiar.
 
-To better understand how this works, let's visualize a simple producer-consumer flow:
+We could go into the technical and theoretical implications of this, but instead lets try a pragmatic approach to really just get it to work.
 
+First, Let's imagine we have a server that constantly emits numbers.
+It starts at the state of the number we give it, then counts up in one from htere onward.
+This is what we would call our producer.
+Each time it emits a number, this is an event, and we want to handle it with a consumer.
+A consumer simply takes what a producer emits and does something to it.
+In our case, we will display the count.
+There is a lot more to GenStage at a technical and applied level, but we will build up on the specifics and definitions further in later lessons, for now we just want a running example we can build up on.
```

Comment:
I guess the motivation here was largely inspired by being incredibly confused on my own first dive into genstage because it just has _so much to it_. I feel that by going straight in and getting something working first, it can be dissected easier and make it simpler to understand the library as a whole once beyond the trivial example we do here.